llama-index-core/llama_index/core/indices/utils.py

```diff
@@ -90,24 +92,31 @@ def default_parse_choice_select_answer_fn(
     answer_lines = answer.split("\n")
     answer_nums = []
     answer_relevances = []
-    for answer_line in answer_lines:
-        line_tokens = answer_line.split(",")
-        if len(line_tokens) != 2:
-            if not raise_error:
+    try:
+        for answer_line in answer_lines:
+            line_tokens = answer_line.split(",")
+            if len(line_tokens) != 2:
+                if not raise_error:
+                    continue
+                else:
+                    raise ValueError(
+                        f"Invalid answer line: {answer_line}. "
+                        "Answer line must be of the form: "
+                        "answer_num: <int>, answer_relevance: <float>"
+                    )
+            answer_num = int(line_tokens[0].split(":")[1].strip())
+            if answer_num > num_choices:
                 continue
-            else:
-                raise ValueError(
-                    f"Invalid answer line: {answer_line}. "
-                    "Answer line must be of the form: "
-                    "answer_num: <int>, answer_relevance: <float>"
-                )
-        answer_num = int(line_tokens[0].split(":")[1].strip())
-        if answer_num > num_choices:
-            continue
-        answer_nums.append(answer_num)
-        # extract just the first digits after the colon.
-        _answer_relevance = re.findall(r"\d+", line_tokens[1].split(":")[1].strip())[0]
-        answer_relevances.append(float(_answer_relevance))
+            answer_nums.append(answer_num)
+            # extract just the first digits after the colon.
+            _answer_relevance = re.findall(
+                r"\d+", line_tokens[1].split(":")[1].strip()
+            )[0]
+            answer_relevances.append(float(_answer_relevance))
+    except Exception as e:
+        _logger.error(f"Error in LLM Rerank: {e!s}")
+        for i in answer_lines:
+            _logger.error(i)
```

Comment:
does this actually fix the error? won't this return an empty array now? 

imo it was probably safer for end users to be raising an error 🤔

Reply:
Its to handle when something like this happened

![image](https://github.com/user-attachments/assets/41f2712a-8f6c-41db-901d-1fe6ca1da1e0)

Reply:
It is a rare case but it's still brok my app. i think, returning what the LLM really says will really help the user to mitigae what kind of issue they're facing, instead of just "list index is out of range"

Reply:
Yes but now we are hiding the error in a log statement, and then returning zero nodes. A user might think everything worked fine if they have logs turned off.

Better to raise an error, and the user can handle try/except. We can make the error better by doing something like 

`raise ValueError("Unable to parse LLM output: {output})`

Reply:
Nice Idea, I'll update the code for sure. Thanks 👍